src/sagemaker/jumpstart/exceptions.py

```diff
@@ -169,3 +195,29 @@ def __init__(
             )
 
         super().__init__(self.message)
+
+
+class MarketplaceModelSubscriptionError(ValueError):
+    """Exception raised when trying to deploy a JumpStart Marketplace model.
+
+    A caller is required to subscribe to the Marketplace product in order to deploy.
+    This exception is raised when a caller tries to deploy a JumpStart Marketplace
+    but the caller is not subscribed to the model.
+    """
+
+    def __init__(
+        self,
+        model_subscription_link: Optional[str] = None,
+        message: Optional[str] = None,
+    ):
+        if message:
+            self.message = message
+        else:
+            if not model_subscription_link:
+                raise RuntimeError("Must specify `model_subscription_link` in arguments.")
```

Comment:
so an error constructor can raise another error?

Isn't this odd? Wouldn't it be better to default to show an error message w/o the subscription link in that case?

Reply:
Agreed, I think that's better